test_runner/regress/test_branching.py

```diff
@@ -128,3 +137,233 @@ def test_branching_unnormalized_start_lsn(neon_simple_env: NeonEnv, pg_bin: PgBi
     endpoint1 = env.endpoints.create_start("b1")
 
     pg_bin.run_capture(["pgbench", "-i", endpoint1.connstr()])
+
+
+def test_cannot_create_endpoint_on_non_uploaded_timeline(neon_env_builder: NeonEnvBuilder):
+    """
+    Endpoint should not be possible to create because branch has not been uploaded.
+    """
+
+    env = neon_env_builder.init_configs()
+    env.start()
+
+    env.pageserver.allowed_errors.append(
+        ".*request{method=POST path=/v1/tenant/.*/timeline request_id=.*}: request was dropped before completing.*"
+    )
+    env.pageserver.allowed_errors.append(
+        ".*page_service_conn_main.*: query handler for 'basebackup .* is not active, state: Loading"
+    )
+    ps_http = env.pageserver.http_client()
+
+    # pause all uploads
+    ps_http.configure_failpoints(("before-upload-index-pausable", "pause"))
+    ps_http.tenant_create(env.initial_tenant)
+
+    initial_branch = "initial_branch"
+
+    with pytest.raises(ReadTimeout):
+        ps_http.timeline_create(env.pg_version, env.initial_tenant, env.initial_timeline, timeout=2)
```

Comment:
Basically all these can stall on ci. They should be cancelled only when a relevant message appears in log.

Reply:
[cb2e4f4](https://github.com/neondatabase/neon/pull/5484/commits/cb2e4f4f8d25fc06f4fb441410dc93c425f13240)